tests/vm_test.go

```diff
@@ -1653,16 +1654,28 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 						newVMI, err := virtClient.VirtualMachineInstance(newVM.Namespace).Get(newVM.Name, &k8smetav1.GetOptions{})
 						Expect(err).ToNot(HaveOccurred())
 						return newVMI.Status.Phase == v1.Failed
-					}, 400*time.Second, 1*time.Second).Should(BeTrue())
+					}, 480*time.Second, 1*time.Second).Should(BeTrue())
 
-					// Virt launcher should be in the Running phase if the Annotation v1.KeepLauncherAfterFailureAnnotation is set to true
-					By(fmt.Sprintf("Verify that the VMI VirtLauncher Pod is in the expected state as %v", podPhase))
+					// The containerStatus of the compute container of the virt-launcher pod should be in the Running state
+					// if the Annotation v1.KeepLauncherAfterFailureAnnotation is set to true
+					By(fmt.Sprintf("Verify that the compute container of the virt-launcher pod is in the running state"))
 					newVMI, _ := virtClient.VirtualMachineInstance(newVM.Namespace).Get(newVM.Name, &k8smetav1.GetOptions{})
 					launcherPod := libvmi.GetPodByVirtualMachineInstance(newVMI, newVM.Namespace)
-					Expect(launcherPod.Status.Phase).To(Equal(podPhase))
+					keepLauncherBool, _ := strconv.ParseBool(keepLauncher)
+					Eventually(Consistently(func() bool {
+						if keepLauncherBool {
+							for _, status := range launcherPod.Status.ContainerStatuses {
+								if status.Name == "compute" && (status.State.Running != nil) == keepLauncherBool {
```

Comment:
`Expect(launcherPod.Status.Phase).To(Equal(podPhase))` was just fine, no need to go look at "compute"

Reply:
In my cluster-up environment, I have consistently run into the situation that the launcher pod is in 'NotReady' phase with 1/2 container in Ready when I check the container with kubectl get pods. This was because the container-disk container ran into completed state and the virt-launcher container was in running state. This could be the reason that the faulty scenario for the test is to create a kernel boot VMI with a mismatched disk.

Reply:
Ah, I see, in that case it's probably fine to focus on compute.

Reply:
@jean-edouard I uploaded another round of change.